tango_simlib/tests/test_tango_sim_generator.py

```diff
@@ -154,3 +162,122 @@ def test_sim_control_device_attribute_change(self):
+        cls.sim_device = tango.DeviceProxy(
+                '%s:%s/test/nodb/tangodeviceserver#dbase=no' % (
+                    cls.host, cls.port))
+        cls.sim_control_device = tango.DeviceProxy(
+                '%s:%s/test/nodb/tangodeviceservercontrol#dbase=no' % (
+                    cls.host, cls.port))
+        cls.addCleanupClass(cls.sub_proc.kill)
+        cls.addCleanupClass(shutil.rmtree, cls.temp_dir)
+
+    def setUp(self):
+        super(test_TangoSimGenDeviceIntegrationForFandangoFile, self).setUp()
+        self.fandango_parser = fandango_json_parser.FandangoExportDeviceParser()
+        self.fandango_parser.parse(self.data_descr_file[0])
+        self.expected_model = tango_sim_generator.configure_device_model(
+                self.data_descr_file, self.sim_device.name())
+        self.attr_name_enum_labels = sorted(
+                self.sim_control_device.attribute_query(
+                     'attribute_name').enum_labels)
+
+    def test_device_attribute_list(self):
+        """ Testing whether the attributes specified in the fandango generated fgo file
+        are added to the TANGO device.
+        """
+        # test that the attributes from the running simulated device match the attributes
+        # from in the fandango generated file
+        device_attributes = set(self.sim_device.get_attribute_list())
+        extra_attr_from_device = set(['NumAttributesNotAdded', 'AttributesNotAdded'])
+        remaining_device_attrs = device_attributes - extra_attr_from_device
+        expected_attributes = []
+        for attr, attr_prop in self.fandango_parser._device_attributes.items():
+            expected_attributes.append(attr_prop['name'])
+
+        self.assertEqual(set(expected_attributes), remaining_device_attrs,
+                         "Actual tango device attribute list differs from expected "
+                         "list!")
+
+    def test_device_command_list(self):
+        """Testing whether commands from running simulated device match commands from
+        fandango file
+        """
+        actual_device_commands = set(self.sim_device.get_command_list())
+        expected_command_list = set(self.fandango_parser.get_device_command_metadata().keys())
+        self.assertEquals(actual_device_commands, expected_command_list,
+                          "The commands specified in the fgo file are not present in"
+                          " the device")
+
+    def _count_device_properties(self):
+        """Count device properties in tango database"""
+        db_info = self.db_instance.get_info()
+        db_info_list = db_info.split('\n')
+        num_properties = 0
+        for line in db_info_list:
+            if 'Device properties defined' in line:
+                num_properties = line.split('=')[-1]
+        return int(num_properties)
+
+    def test_initial_device_properties(self):
+        """Test initial device properties added to the tangoDB"""
+        expected_count = 1
```

Comment:
The code doesn't make it clear which property we are expecting.  The test this was copied from had a comment `# model_key property already present in db`.  

- [ ] Suggest that is used here as well (if that is indeed the same property we are getting, if not, provide a relevant comment).